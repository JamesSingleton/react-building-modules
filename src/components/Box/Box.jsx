import styled from 'styled-components';
import { space, color, layout } from 'styled-system';

const Box = styled.div(
  {
    boxSizing: 'border-box',
    minWidth: 0,
  },
  space,
  color,
  layout
);

export default Box;
