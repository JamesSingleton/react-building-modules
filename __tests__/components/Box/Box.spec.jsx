import React from 'react';
import { shallow } from 'enzyme';
import Box from '../../../src/components/Box';

describe('Box', () => {
  it('should render as a component', () => {
    const renderedButton = shallow(<Box />);
    expect(renderedButton).toMatchSnapshot();
  });
});
