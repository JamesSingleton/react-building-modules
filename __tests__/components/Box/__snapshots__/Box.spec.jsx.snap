// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render as a component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          "min-width: 0;",
          [Function],
          [Function],
          [Function],
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
